Lima: Use OpenRC to manage dockerd lifetime.

mook-as opened the pull request:
This changes the K3s OpenRC init script to depend on docker (when using the moby backend), so that we can automatically start/stop it via OpenRC instead of manually managing its lifetime.  This ensures correct start up and shutdown ordering.

Also update the WSL side to use the same service name ("docker" rather than "dockerd").

Fixes #1009

mook-as commented:
We will need a follow-up PR to port some of the changes from #1048 over after this is merged.

ericpromislow reviewed (changes requested):
Doesn't work:

Images aren't shown.  In `images.log`:

```
2021-12-02T03:07:20.851Z: error during connect: Get
 "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/json": EOF
...
```

On the VM:
```
$ docker images
Got permission denied while trying to connect to the Docker daemon socket at 
unix:///var/run/docker.sock: 
Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/json": 
dial unix /var/run/docker.sock: connect: permission denied

$ ls -l /var/run/docker.sock
srw-rw----    1 root     docker           0 Dec  2 03:10 /var/run/docker.sock
```

I thought we were going to make that file owned by the lima user. Possibly something got dropped when moving to `openrc`

mook-as commented:
Right, I dropped some code that made the socket world-writable.  Fixed by actually adding the lima user to the `docker` group instead, which seems like a better fix (no need to wait for the socket to exist, matches normal Linux distros).

I also picked up a fix from #1048 where k3s may or may not depend on docker.

ericpromislow reviewed (approved):
Works fine now, LGTM
